site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+  - kind: Service
+    name: foo
+  rules:
+    backendRefs:
+    - kind: Service
+      name: foo
+      weight: 90
+    - kind: Service
+      name: foo_v2
+      weight: 10
+```
+
+In the example above, routing rules have been configured to direct 90% of traffic for the `foo` `Service` to the default "backend" endpoints specified by the `foo` `Service` [`selector`](https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service) field, and 10% to the `foo_v2` `Service`. This is determined based on the `ClusterIP` (for `Service`) and `ClusterSetIP` (for `ServiceImport`) matching, and for "transparent proxy" mesh implementations would match all requests to `foo.svc.cluster.local` (or arbitrary custom suffix, as the hostname is not specified manually) from within the same namespace, all requests to `foo.store.svc.cluster.local` from other namespaces, and all requests to `foo.store.svc.clusterset.local` for multicluster services, within the scope of the service mesh.
+
+### Omitting `backendRefs`
+
+Implementations SHOULD support a terse syntax which allows omitting `backendRefs` to avoid unnecessary redundancy for simple configurations. If no `backendRefs` are specified, implementations should direct traffic to the default endpoints of the `parentRef` `Service`. If the `parentRef` does not have any endpoints (e.g. if it is a `Service` without selectors), implementations MUST return an HTTP 503 response code (see discussion in [#1210](https://github.com/kubernetes-sigs/gateway-api/pull/1210)).
+
+This syntax has limited utility currently, but could become more relevant if additional functionality beyond traffic splitting is added to the `HTTPRoute` resource. `HTTPRoutes` configured in this way MAY be used to manually enroll services into a mesh (which could trigger behavior like injecting a sidecar proxy) if they have not already been enrolled by some other mechanism (as proposed in [GEP-1291: Mesh Representation](https://docs.google.com/document/d/1oyA9uUH7pNNxxwy3WZGSWx-edHDBLrujcezr8q3el70/edit) or similar).
+
+```
+metadata:
+  name: foo_route
+  namespace: store
+spec:
+  parentRefs:
+  - kind: Service
+    name: foo
+  rules:
+    matches:
+    - path:
+        value: "/bar"
+```
+
+The example above would drop all incoming traffic for HTTP paths other than `/bar` to the `foo` `Service`, following the existing spec to return HTTP 404 response codes for unmatched requests, and HTTP 500 response codes for requests excluded due to an [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter).
+
+### Allowed `Service` types
+
+Valid `Service` resources to be selected as a `parentRef` SHOULD provision one or more virtual IP(s) and DNS hostname(s) within the Kubernetes cluster, and include the default [`ClusterIP`](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types) `Service` type, with or without selectors.
```

Comment:
```suggestion
Valid `service` resources to be selected as a `parentRef` SHOULD provision one or more virtual IP(s) and DNS hostname(s) within the Kubernetes cluster, and include the default [`ClusterIP`](https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types) `Service` type, with or without selectors.
```

anywhere we are refering to the bag of valid "services" we should probably not use `Service`?

Reply:
But besides terminology - do we want to be that prescriptive, or just that that "services should have a way to identify traffic to them (typically by a virtual IP or name)".

for example, we may have a `S3BucketRoute` (weird, but proves a point), that is identified by the name of the bucket - not an IP or DNS name. The important thing is a proxy can identify the traffic, not _how_

Reply:
I'm in favor of the idea that the services just need a way to identify traffic - Cilium does this slightly differently, and that will cover the way Cilium security identities work.

I'm okay with saying that only Cluster IP is valid _for now_ though.